src/controller/investments.js

```diff
@@ -1,15 +1,22 @@
 const express = require('express');
-const userInvestments = require('../models/data');
 
 const investments = express.Router();
+const { User } = require('../models');
 
-investments.get('/user/:uid', (req, res) => {
-  const { uid } = req.params;
-  const userData = userInvestments.find((data) => data.userId === uid);
-  if (userData) {
-    return res.send(userData);
+investments.get('/user', async (req, res) => {
+  try {
+    const { _id } = req.user;
+    const user = await User.findById(_id, 'investments');
+    if (user) {
+      return res.send(user);
+    }
+    return res.sendStatus(404);
```

Comment:
how can we get 404 with a valid token? We generate the token during login, which means there _is_ a user with this id.